components/brave_wallet/browser/ethereum_provider_impl.cc

```diff
@@ -1664,4 +1703,21 @@ void EthereumProviderImpl::OnGetBlockByNumber(
 
 void EthereumProviderImpl::OnNewBlock(uint256_t block_num) {}
 
+void EthereumProviderImpl::OnLogsReceived(base::Value rawlogs) {
+  if (!rawlogs.is_dict() || !events_listener_.is_bound()) {
+    return;
+  }
+
+  auto& dict = rawlogs.GetDict();
+  const base::Value::List* results = dict.FindList("result");
+
+  for (auto& results_item : *results) {
+    base::ranges::for_each(
+        eth_log_subscriptions_,
+        [this, &results_item](const std::string& subscription_id) {
+          events_listener_->MessageEvent(subscription_id, results_item.Clone());
+        });
```

Comment:
I'm curious why not just a two-liner here? :)
```
for (const auto& subscription_id: eth_log_subscriptions_)
  events_listener_->MessageEvent(subscription_id, results_item.Clone());
```

Reply:
fixed